processor/datadogsemanticsprocessor/processor.go

```diff
@@ -0,0 +1,130 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadogsemanticsprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/datadogsemanticsprocessor"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/traceutil"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/collector/semconv/v1.27.0"
+)
+
+func insertAttrIfMissing(sattr pcommon.Map, key string, value any) {
+	if _, ok := sattr.Get(key); !ok {
+		switch v := value.(type) {
+		case string:
+			sattr.PutStr(key, v)
+		case int64:
+			sattr.PutInt(key, v)
+		}
```

Comment:
Return an error if value is of other types (otherwise this may be a pain to debug in the future)